# installazione librerie necessarie
pandas, numpy, scikit-learn

In [ ]:
!pip install scikit-learn
!pip install numpy
!pip install pandas

# Elaborazione dati
* Dò uno sguardo visivo ai dati csv 

In [ ]:
import pandas as pd

df = pd.read_csv("../data/train.csv")
df.describe().T     # mi risulta più facile leggere i dati trasposti

count       mean        std   min  \
Basic_Demos-Age                         3960.0  10.433586   3.574648   5.0   
Basic_Demos-Sex                         3960.0   0.372727   0.483591   0.0   
CGAS-CGAS_Score                         2421.0  65.454771  22.341862  25.0   
Physical-BMI                            3022.0  19.331929   5.113934   0.0   
Physical-Height                         3027.0  55.946713   7.473764  33.0   
...                                        ...        ...        ...   ...   
PCIAT-PCIAT_Total                       2736.0  27.896199  20.338853   0.0   
SDS-SDS_Total_Raw                       2609.0  41.088923  10.427433  17.0   
SDS-SDS_Total_T                         2606.0  57.763622  13.196091  38.0   
PreInt_EduHx-computerinternet_hoursday  3301.0   1.060588   1.094875   0.0   
sii                                     2736.0   0.580409   0.771122   0.0   

                                             25%        50%        75%  \
Basic_Demos-Age                          8.00000  10.000000  13.000000   
Basic_Demos-Sex                          0.00000   0.000000   1.000000   
CGAS-CGAS_Score                         59.00000  65.000000  75.000000   
Physical-BMI                            15.86935  17.937682  21.571244   
Physical-Height                         50.00000  55.000000  62.000000   
...                                          ...        ...        ...   
PCIAT-PCIAT_Total                       12.00000  26.000000  41.000000   
SDS-SDS_Total_Raw                       33.00000  39.000000  46.000000   
SDS-SDS_Total_T                         47.00000  55.000000  64.000000   
PreInt_EduHx-computerinternet_hoursday   0.00000   1.000000   2.000000   
sii                                      0.00000   0.000000   1.000000   

                                               max  
Basic_Demos-Age                          22.000000  
Basic_Demos-Sex                           1.000000  
CGAS-CGAS_Score                         999.000000  
Physical-BMI                             59.132048  
Physical-Height                          78.500000  
...                                            ...  
PCIAT-PCIAT_Total                        93.000000  
SDS-SDS_Total_Raw                        96.000000  
SDS-SDS_Total_T                         100.000000  
PreInt_EduHx-computerinternet_hoursday    3.000000  
sii                                       3.000000  

[70 rows x 8 columns]

In [9]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3960 entries, 0 to 3959
Data columns (total 82 columns):
 #   Column                                  Non-Null Count  Dtype  
---  ------                                  --------------  -----  
 0   id                                      3960 non-null   object 
 1   Basic_Demos-Enroll_Season               3960 non-null   object 
 2   Basic_Demos-Age                         3960 non-null   int64  
 3   Basic_Demos-Sex                         3960 non-null   int64  
 4   CGAS-Season                             2555 non-null   object 
 5   CGAS-CGAS_Score                         2421 non-null   float64
 6   Physical-Season                         3310 non-null   object 
 7   Physical-BMI                            3022 non-null   float64
 8   Physical-Height                         3027 non-null   float64
 9   Physical-Weight                         3076 non-null   float64
 10  Physical-Waist_Circumference            898 non-null    floa

noto subito che le colonne `PAQ_A-Season` e `PAQ_A-PAQ_A_Total` hanno solo 475 dati non nulli contro i 3960 totali. Similmente, sono pochi i dati disponibili delle colonne `Fitness_Endurance-Time_Mins`, `Fitness_Endurance-Max_Stage`, `Fitness_Endurance-Time_Mins` e `Physical-Waist_Circumference`

* divido in train e set, 70/30


In [16]:
from sklearn.model_selection import train_test_split

# rimuovo le etichette delle classi
x = df.drop(columns=["id", "sii"])
y = df["sii"]
X_train, X_test, y_train, y_test = train_test_split(x, y, test_size=0.3)